packages/eslint-plugin/rules.d.ts

```diff
@@ -62,16 +62,25 @@ export interface ESLintPluginDocs {
   requiresTypeChecking?: boolean;
 }
 
-export type ESLintPluginRuleModule = RuleModuleWithMetaDocs<
+type ESLintPluginRuleModule = RuleModuleWithMetaDocs<
   string,
   readonly unknown[],
   ESLintPluginDocs
 >;
 
-export type TypeScriptESLintRules = Record<
+type TypeScriptESLintRules = Record<
   string,
   RuleModuleWithMetaDocs<string, unknown[], ESLintPluginDocs>
 >;
 
 declare const rules: TypeScriptESLintRules;
-export default rules;
```

Comment:
Why did you switch this from an `export default`?

Reply:
With a `default` export we get:

![tseslint-rules-with-default-export](https://github.com/user-attachments/assets/20596e78-5c79-4701-9c40-979c8169b074)

Which basically means in ESM files we will have an extra `.default` property, so for exmaple if you have an ESM file like `foo.mjs` and you try to import `@typescript-eslint/eslint-plugin/use-at-your-own-risk/rules`:

```js
import rules from '@typescript-eslint/eslint-plugin/use-at-your-own-risk/rules'
console.log(Object.keys(rules)) // We will get [ 'default' ]
```

The `console.log(Object.keys(rules))` should give us all the rules but instead we get `[ 'default' ]` which is not correct.